python/ray/experimental/sgd/interaces.py

```diff
@@ -0,0 +1,134 @@
+from dataclasses import dataclass
+
+from typing import Callable
+
+
+@dataclass
+class WorkerArgs:
+    num_workers: int
+    num_cpus_per_worker: int
+    num_gpus_per_worker: int
+    placement_strategy: str # Assumes 1 bundle per worker.
+
+    elastic_training: bool = False # If True, training will continue with fewer
+    # workers if an actor fails. Also, if more resources are available,
+    # training will continue with more workers.
+    max_actor_restarts: int = 0 # Number of retries when Ray actor fails.
+    # Doesn't do anything if elastic_training is True. Set to -1 for
+    # unlimited restarts.
+    max_failed_actors: int = 0 # If `elastic_training` is True, this
+            # specifies the maximum number of failed actors with which
+            # we still continue training.
+
+class BackendConfig:
+    pass
+
+@dataclass
+class TorchConfig(BackendConfig):
+    address: str # The connection address to use for the process group.
+    process_group_timeout: int # The timeout for process group creation.
+
+@dataclass
+class TFConfig(BackendConfig):
+    pass
+
+
+def setup_torch_process_group(...):
+    pass
+
+def shutdown_torch(...):
+    pass
```

Comment:
I'm fine either way, though these don't need to access any state from `TorchExecutor`, right?

Reply:
Oh, woops I commented this at first since everything is in one python file. More of a personal note to define these in the same file as the `TorchExecutor`.